doc/api/diagnostics_channel.md

```diff
@@ -407,6 +457,526 @@ channel.subscribe(onMessage);
 channel.unsubscribe(onMessage);
 ```
 
+#### `channel.bindStore(store[, transform])`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `store` {AsyncLocalStorage} The store to which to bind the event data
+* `transform` {Function} Transform event data before setting the store context
+
+When [`channel.runStores(data, ...)`][] is called, the given event data will
+be applied to any store bound to the channel. If the store has already been
+bound the previous `transform` function will be replaced with the new one.
+The `transform` function may be omitted to have the event data be used as the
+context directly.
+
+```mjs
+import diagnostics_channel from 'node:diagnostics_channel';
+import { AsyncLocalStorage } from 'node:async_hooks';
+
+const store = new AsyncLocalStorage();
+
+const channel = diagnostics_channel.channel('my-channel');
+
+channel.bindStore(store, (data) => {
+  return { data };
+});
+```
+
+```cjs
+const diagnostics_channel = require('node:diagnostics_channel');
+const { AsyncLocalStorage } = require('node:async_hooks');
+
+const store = new AsyncLocalStorage();
+
+const channel = diagnostics_channel.channel('my-channel');
+
+channel.bindStore(store, (data) => {
+  return { data };
+});
+```
+
+#### `channel.unbindStore(store)`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `store` {AsyncLocalStorage} The store to unbind from the channel.
+* Returns: {boolean} `true` if the store was found, `false` otherwise.
+
+Remove a message handler previously registered to this channel with
+[`channel.bindStore(store)`][].
+
+```mjs
+import diagnostics_channel from 'node:diagnostics_channel';
+import { AsyncLocalStorage } from 'node:async_hooks';
+
+const store = new AsyncLocalStorage();
+
+const channel = diagnostics_channel.channel('my-channel');
+
+channel.bindStore(store);
+channel.unbindStore(store);
+```
+
+```cjs
+const diagnostics_channel = require('node:diagnostics_channel');
+const { AsyncLocalStorage } = require('node:async_hooks');
+
+const store = new AsyncLocalStorage();
+
+const channel = diagnostics_channel.channel('my-channel');
+
+channel.bindStore(store);
+channel.unbindStore(store);
+```
+
+#### `channel.runStores(data, fn[, thisArg[, ...args]])`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `data` {any} Message to send to subscribers and bind to stores
+* `fn` {Function} Handler to run within the entered storage context
+* `thisArg` {any} The receiver to be used for the function call.
+* `...args` {any} Optional arguments to pass to the function.
+
+Publishes data to the channel and applies it to any AsyncLocalStorage instances
```

Comment:
I think we should document the sequence that first publish happens and then the store is entered.

Reply:
Could do that. Can you think of any examples when that ordering would matter? Do we need to say anything beyond just swapping that `and` for a `then`?

Reply:
It should be clear what one gets if  `als.getStore()` is called inside the `onMessage` function. Same is valid for the `transform` function where sample relies on the fact that it happens before `als.run()` is called to get the parent.

I think it's important to tell users that `publish` and `transform` happen before.

Reply:
Fair. I'll make some improvements to that soon. :)